File: lesson3/storage.py
import json
import os

STORAGE_FILE = 'topics.json'

def save_topics(chat_id, topics):
    data = {}
    if os.path.exists(STORAGE_FILE):
        with open(STORAGE_FILE, 'r') as f:
            data = json.load(f)
    data[str(chat_id)] = topics
    with open(STORAGE_FILE, 'w') as f:
        json.dump(data, f)

def get_topics(chat_id):
    if not os.path.exists(STORAGE_FILE):
        return {}
    with open(STORAGE_FILE, 'r') as f:
        data = json.load(f)
    return data.get(str(chat_id), {})

File: agents/storage.js
const fs = require('fs');
const path = require('path');

const TOPICS_FILE = path.join(__dirname, 'topics.json');

function load() {
  if (fs.existsSync(TOPICS_FILE)) {
    return JSON.parse(fs.readFileSync(TOPICS_FILE, 'utf8'));
  }
  return {};
}

function save(data) {
  fs.writeFileSync(TOPICS_FILE, JSON.stringify(data, null, 2), 'utf8');
}

let data = load();

function saveTopics(chatId, topics) {
  data[String(chatId)] = topics;
  save(data);
}

function getTopics(chatId) {
  return data[String(chatId)] || {};
}

module.exports = { saveTopics, getTopics };
